src/storage/gpu_swap_memmgr.cc

```diff
@@ -77,7 +77,10 @@ bool CudaMemoryManager::TryAllocate(int device_id, size_t size) {
   size_t free, total;
   CUDA_CALL(cudaMemGetInfo(&free, &total));
   // TODO(fegin): This fixed threshold is not acceptable.
-  return free > size + 1000000000;
+  // FIXME(fegin): The maximum threshould I used in the old MXNet is 512 MB.
+  //               We should figure out why such a large threshold is needed
+  //               for current implementation.
```

Comment:
Did 512 MB work with all size of model on 8GB gpu?

Reply:
Mm, I did the tests on 12GB GPU. However, the threshold should be for the internal usage of CUDA (like cuDNN) PLUS memory fragments. Unless memory fragments cause 1.5GB , which is unlikely, I don't think the GPU's size matters that much (512MB v.s 1.5GB). But I'm not sure. It worth to investigate.

Reply:
True. Also 1.5G is not a tight limit. I just increased 1G to 1.5G because 1G was not working.